src/main/java/pl/ergohestia/ehj1/ivesta/ui/DriverMenu.java

```diff
@@ -0,0 +1,145 @@
+package pl.ergohestia.ehj1.ivesta.ui;
+
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.ergohestia.ehj1.ivesta.model.DriverDto;
+import pl.ergohestia.ehj1.ivesta.model.Menu;
+import pl.ergohestia.ehj1.ivesta.services.DriverService;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Scanner;
+
+@Slf4j
+public class DriverMenu {
+
+    private Logger SYSOUT = LoggerFactory.getLogger("SYSOUT");
+    private InputScannerProvider in = new InputScannerProvider();
+
+    DriverService driverService = new DriverService();
+
+    public void runDriverMenu() {
+        serviceDriverMenu();
+    }
+
+    private final Menu driverMenu = new Menu(
+            "1. Wyświetl wszystkich kierowców.",
+            "2. Wyświetl dostępnych kierowców.",
+            "3. Edytuj kierowców.",
+            "4. Załaduj nowych kierowców.",
+            "5. Wróć do głównego menu.");
+
+    private void subMenuShowAllDrivers() {
+        logSubMenu(1);
+        List<DriverDto> list = driverService.getAllDrivers();
+        if (!(list == null) && !list.isEmpty()) {
+            list.stream().map(DriverDto::toString).forEach(SYSOUT::info);
+        } else {
+            SYSOUT.info("Brak kierowców. Dodaj nowych.");
+        }
+    }
+
+    private void subMenuShowAvailableDrivers() {
+        logSubMenu(2);
+        List<DriverDto> list = driverService.getAvailableDrivers();
+        if (!(list == null) && !list.isEmpty()) {
+            list.stream().map(DriverDto::toString).forEach(SYSOUT::info);
+        } else {
+            SYSOUT.info("Brak dostępnych kierowców.");
+        }
+    }
+
+    private void subMenuEditDrivers() {
+        logSubMenu(3);
+        List<DriverDto> list = driverService.getAllDrivers();
+        if (!(list == null) && !list.isEmpty()) {
+            driverService.editDriversData(list);
+        } else {
+            SYSOUT.info("Brak kierowców. Dodaj nowych.");
+        }
+    }
+
+    private void subMenuLoadNewDrivers() {
+        logSubMenu(4);
+        driverService.setDriverPath(getDriverFilePathFromInput());
+        driverService.importDrivers();
+    }
+
+    private void logSubMenu(int index) {
+        int menuIndex = index - 1;
+        String subMenuName = driverMenu.getMenuItems().get(menuIndex);
+        SYSOUT.info("Wybrałeś: " + subMenuName);
+        log.info("User correctly chose " + index + " in menu");
+    }
+
+    private void serviceDriverMenu() {
+        int item = 0;
+        while (item != 5) {
+            printMenu(driverMenu.getMenuItems());
+            item = getMenuItem();
+            switch (item) {
+                case 1 -> subMenuShowAllDrivers();
+                case 2 -> subMenuShowAvailableDrivers();
+                case 3 -> subMenuEditDrivers();
+                case 4 -> subMenuLoadNewDrivers();
+                case 5 -> logSubMenu(5);
```

Comment:
to nie wraca do głównego Menu. Ja w najnowszej wersji zrobiłem to tak że: W MenuService w twoim przypadku przekazał bym this tzn 
driverMenu.runDriverMenu(this);
wtedy w DriverMenu metodę runDriverMenu(MenuService mainMenu), przypisać main Menu do MenuService, i jako punkt piąty wywołać mainMenu.menu()

Takie coś możesz zobaczyć w nowej wersji developa w MenuService tutaj:
    private void subMenuVehicle() {
        logSubMenu(2);
        vehicleMenu.runVehicleMenu(**this**);
    }

Reply:
Byłem pewien, że w MenuService główne menu będzie zapętlone, dlatego tak to zrobiłem. Pętla działa póki użytkownik nie wybierze opcji powrotu do głównego menu i można sobie najpierw wczytać kierowców, potem ich edytować, potem ich wyświetlić i tak póki ci się to submenu nie znudzi.

Patrzyłem na twoje rozwiązanie i wydaje się trochę niepotrzebnie skomplikowane.

Spójrz proszę na moją propozycję (wrzuciłem kolejny commit) , gdzie wystarczy trochę przerobić dwie metody w MenuService. Moim zdaniem tak jest prościej i dodatkowo można zrobić pętle w dodatkowych submenu tak jak u mnie w DriverMenu, dzięki czemu po wykonaniu jakiegoś działania w submenu, nie cofa cię do głównego menu dopóki nie wybierzesz takiej opcji.

Minus jest taki, że trzeba by przerobić VehicleMenu, bo po moich zmianach dalej działa, ale dwa razy printuje opcje z głównego menu.

Reply:
serviceVehicleMenu poprawiłem na coś takiego. Działa w pętli a po wyjściu wywołuje menuService.menu

![image](https://user-images.githubusercontent.com/44324356/162609721-eecb294f-9ceb-434b-b97e-364597b1dd99.png)

Reply:
I wtedy cały develop zostaje tak jak jest teraz, a w runVehicleMenu zostaje tylko to:
![image](https://user-images.githubusercontent.com/44324356/162609845-9a9424ed-42ce-403e-945d-67a6492d516c.png)

Reply:
Analogicznie zrobiłbym dla kierowców